### PR TITLE
Add space after '?' only for conditional operators

### LINKED ISSUE #2080: Formatter inserts extra space after ? in optional signatures
Type the following:
```ts
interface I {
    x?(): any;
    y?<T>(): any;
}
```

Upon formatting, notice the extra space after the ?. It should not be there. Ditto for generic signatures.

### PR DESCRIPTION
**Pull request by vladima**
Fixes #2080

### CODE REVIEW

File: src/services/formatting/rules.ts [hunk 1]
```diff
@@ -217,7 +221,8 @@ module ts.formatting {
             this.NoSpaceBeforeColon = new Rule(RuleDescriptor.create2(Shared.TokenRange.Any, SyntaxKind.ColonToken), RuleOperation.create2(new RuleOperationContext(Rules.IsSameLineTokenContext, Rules.IsNotBinaryOpContext), RuleAction.Delete));
             this.NoSpaceBeforeQMark = new Rule(RuleDescriptor.create2(Shared.TokenRange.Any, SyntaxKind.QuestionToken), RuleOperation.create2(new RuleOperationContext(Rules.IsSameLineTokenContext, Rules.IsNotBinaryOpContext), RuleAction.Delete));
             this.SpaceAfterColon = new Rule(RuleDescriptor.create3(SyntaxKind.ColonToken, Shared.TokenRange.Any), RuleOperation.create2(new RuleOperationContext(Rules.IsSameLineTokenContext, Rules.IsNotBinaryOpContext), RuleAction.Space));
-            this.SpaceAfterQMark = new Rule(RuleDescriptor.create3(SyntaxKind.QuestionToken, Shared.TokenRange.Any), RuleOperation.create2(new RuleOperationContext(Rules.IsSameLineTokenContext, Rules.IsNotBinaryOpContext), RuleAction.Space));
```
Comment (DanielRosenwasser) [inline src/services/formatting/rules.ts]: I don't understand; why didn't this catch that originally?

File: src/services/formatting/rules.ts [hunk 2]
```diff
@@ -37,7 +37,11 @@ module ts.formatting {
         public NoSpaceBeforeColon: Rule;
         public NoSpaceBeforeQMark: Rule;
         public SpaceAfterColon: Rule;
-        public SpaceAfterQMark: Rule;
+        // insert space after '?' only when it is used in conditional operator
+        public SpaceAfterQMarkInConditionalOperator: Rule;
+        // in other cases there should be no space between '?' and next token
+        public NoSpaceAfterQMark: Rule;
```
Comment (CyrusNajmabadi) [inline src/services/formatting/rules.ts]: Don't abbreviate QuestionMark to QMark.

:+1:

File: tests/cases/fourslash/formattingConditionalOperator.ts [hunk 3]
```diff
@@ -0,0 +1,6 @@
+/// <reference path='fourslash.ts'/>
+
+////var x=true?1:2
+format.document();
+goTo.bof();
+verify.currentLineContentIs("var x = true ? 1 : 2");;
```
Comment (DanielRosenwasser) [inline tests/cases/fourslash/formattingConditionalOperator.ts]: `;;` not that it really matters

**Comment by DanielRosenwasser** [comment]
:+1:

**Comment by JsonFreeman** [comment]
:+1:

**Pull request merged**